src/app/components/search/search.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { debounceTime, filter, tap } from 'rxjs';
+import { SearchApiService } from './search-api.service';
+import { SearchResponse } from './types/search-response';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'mdb-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.scss'],
+})
+export class SearchComponent implements OnInit {
+  constructor(private searchApi: SearchApiService) {}
+
+  public searchInput = new FormControl();
+  public searchResult?: SearchResponse;
+  public debounceTime = 200;
+  public minSearchLength = 3;
+
+  public clear() {
+    this.searchInput.setValue('');
+    this.searchResult = undefined;
+  }
+
+  public ngOnInit(): void {
+    this.searchInput.valueChanges
+      .pipe(
+        tap(() => (this.searchResult = undefined)),
+        debounceTime(this.debounceTime),
+        filter(val => val.length >= this.minSearchLength)
+      )
+      .subscribe(searchString => {
+        this.searchApi
+          .requestSearchResult(searchString)
+          .subscribe(result => (this.searchResult = result));
```

Comment:
[Nested](https://www.thinktecture.com/en/angular/rxjs-antipattern-1-nested-subs/) `subscribe` is considered as not good practice. This approach leads us to callback hell. Please, use [Higher-orderer Observable](https://blog.angular-university.io/rxjs-higher-order-mapping/)

Reply:
Reworked component to avoid nested subscriptions.